services/core-api/app/extensions.py

```diff
@@ -1,14 +1,54 @@
 import logging
+import os
 
 from flask_caching import Cache
-from flask_jwt_oidc import JwtManager
+from flask import Flask
+from app.flask_jwt_oidc_local.exceptions import AuthError
+from app.flask_jwt_oidc_local import JwtManager
 from flask_sqlalchemy import SQLAlchemy
-
+from app.config import TestConfig
+from jose import jwt as jwt_jose
 from .config import Config
 from .helper import Api
 
+def JWT_ROLE_CALLBACK(jwt_dict):
+    return (jwt_dict.get('client_roles') or [])
+
+def JWT_ROLE_CALLBACK_V1(jwt_dict):
+        return (jwt_dict['realm_access']['roles'])
+
 db = SQLAlchemy()
-jwt = JwtManager()
+
+# Gold SSO
+jwtv2 = JwtManager(None, os.environ.get('JWT_OIDC_WELL_KNOWN_CONFIG'), None, 'RS256', None, None, os.environ.get('JWT_OIDC_AUDIENCE'), None, None, False, False, None, JWT_ROLE_CALLBACK, None)
+# Existing Keycloak for integration clients
+jwtv1 = JwtManager(None, os.environ.get('JWT_OIDC_WELL_KNOWN_CONFIG_V1'), None, 'RS256', None, None, os.environ.get('JWT_OIDC_AUDIENCE_V1'), None, None, False, False, None, JWT_ROLE_CALLBACK_V1, None)
+# Test JWT Config for integration tests
+test_config = TestConfig()
+jwt = JwtManager(None, test_config.JWT_OIDC_WELL_KNOWN_CONFIG, None, 'RS256', None, None, test_config.JWT_OIDC_TEST_AUDIENCE, None, None, False, True, test_config.JWT_OIDC_TEST_KEYS, JWT_ROLE_CALLBACK, test_config.JWT_OIDC_TEST_PRIVATE_KEY_PEM)
+
+def getJwtManager():
```

Comment:
picks jwt obj from above based on the API request, applies only to request context. 

objects remain in the app context.